hack/gha-reversemap.sh

```diff
@@ -0,0 +1,260 @@
+    yq ".${action_ref}.sha = \"${action_sha}\"" -i $REVERSEMAP_FILE
+    yq ".${action_ref}.sha-url = \"https://github.com/${action_ref_safe}/commit/${action_sha}\"" -i $REVERSEMAP_FILE
+    yq ".${action_ref}.tag = \"${action_tag}\"" -i $REVERSEMAP_FILE
+    yq ".${action_ref}.tag-url = \"https://github.com/${action_ref_safe}/tree/${action_tag}\"" -i $REVERSEMAP_FILE
+}
+
+# Synchronise reverse map values with an action full ref (ie. actions/checkout@v4.2.0)
+_sync_reversemap_with() {
+    filename=$1
+    for action_fullref in $(yq '.jobs[].steps[] | select(has("uses")) | .uses ' "$filename"); do
+        action_ref=$(echo "$action_fullref" | cut -d '@' -f 1)
+        action_tag=$(echo "$action_fullref" | cut -d '@' -f 2)
+        length=${#action_tag}
+        _loginfo "ref=$action_ref and tag=$action_tag len=$length"
+        if [[ $length -ne $GIT_COMMITSHA_LENGTH ]]; then
+            action_sha=$(_fetch_sha_from_upstream_ref "$action_ref" "$action_tag")
+            _loginfo "taking commit of action=${action_ref} tag=${action_tag} sha=${action_sha}"
+            _yq_update_reversemap "$action_ref" "$action_tag" "$action_sha"
+        fi
+    done
+}
+
+# Fetch latest release tag of an action upstream
+_fetch_latest_tag() {
+    action_ref=$1
+    API_GITHUB_LATEST_RELEASE=https://api.github.com/repos/$action_ref/releases/latest
+    _return "$(curl -s -H "Authorization: Bearer $GITHUB_TOKEN" "$API_GITHUB_LATEST_RELEASE" | jq -r '.tag_name')"
+}
+
+# Update action version within a file
+_update_action_version_infile() {
+    file=$1
+    action_ref=$2
+    action_sha=$3
+    sed -E -i '' "s;$action_ref@.+;$action_ref@$action_sha;g" "$file"
+}
+
+# Get commit sha of an action ref stored in the reverse map
+_get_sha_from_reversemap() {
+    action_ref=$1
+    query=$(yq ".\"$action_ref\".sha" $REVERSEMAP_FILE)
+    if [[ $? -ne 0 ]]; then
+        _error_exit "$query"
+    fi
+    _return "$query"
+}
+
+# Run operation apply
+run_apply() {
+    for file in "$GITHUB_WORKFLOWS_PATH"/*.yml; do
+        _loginfo "applying '$REVERSEMAP_FILE' commit sha to be used in '$file' ..."
+        action_refs=$(grep -E -o "$RE_ACTION_REF_AT" "$file")
+        for action_ref in $action_refs; do
+            action_ref=${action_ref%?}
+            action_sha=$(_get_sha_from_reversemap "$action_ref")
+            _loginfo "found $action_ref in reversemap with sha=$action_sha"
+            _update_action_version_infile "$file" "$action_ref" "$action_sha"
+        done
+    done
+}
+
+# Run operation update
+run_update() {
+    dependency_arg=$1
+    _loginfo "updating dependency '$dependency_arg' tag to latest version available inside reverse map '$REVERSEMAP_FILE'"
+    latest_tag=$(_fetch_latest_tag "$dependency_arg")
+    action_sha=$(_fetch_sha_from_upstream_ref "$dependency_arg" "$latest_tag")
+    _yq_update_reversemap "$dependency_arg" "$latest_tag" "$action_sha"
+}
+
+# Run operation sync
```

Comment:
I would use a little more expansive comment, perhaps something like "updates the reverse map file based on the tags/branches currently referenced in all the workflows".